shotover/src/transforms/throttling.rs

```diff
@@ -37,6 +39,7 @@ impl TransformConfig for RequestThrottlingConfig {
 pub struct RequestThrottling {
     limiter: Arc<RateLimiter<NotKeyed, InMemoryState, DefaultClock, NoOpMiddleware>>,
     max_requests_per_second: NonZeroU32,
+    throttled_requests: HashMap<MessageId, Message>,
```

Comment:
Aren't MessageId the random 128Bit numbers? 
So, collisions should not be a problem and we could avoid hashing and having a single table ID -> Message?

My view is that, although this looks like a weird optimization step, I think if we are looking at performance, all these trivial things might start to add up under very heavy load. Hashing is a fast operation in today CPUs but if we can skip those, why not?

Reply:
I'm not sure what you are suggesting. A `Vec<(MessageId, Message>)` to do a linear look up instead of hashmap?